src/cargo/ops/lockfile.rs

```diff
@@ -96,15 +96,20 @@ fn are_equal_lockfiles(mut orig: String, current: &str, lock_update_allowed: boo
         orig = orig.replace("\r\n", "\n");
     }
 
-    // Old lockfiles have unused `[root]` section,
-    // just ignore it if we are in the `--frozen` mode.
-    if !lock_update_allowed && orig.starts_with("[root]") {
-        orig = orig.replacen("[root]", "[[package]]", 1);
-        match (orig.parse::<toml::Value>(), current.parse::<toml::Value>()) {
```

Comment:
😭 

I've checked that `toml::Value` uses `BTree` for maps, but of course it is irrelevant here, because `[[package]]` is an array, and not a map.